File: Examples/VisualBasic/ConvertingCAD/ListLayouts.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports Aspose.CAD
Namespace ConvertingCAD
    Public Class ListLayouts
        Public Shared Sub Run()
            ' ExStart:ListLayouts
            ' The path to the documents directory.
            Dim MyDir As String = RunExamples.GetDataDir_ConvertingCAD()
            Dim sourceFilePath As String = MyDir & Convert.ToString("conic_pyramid.dxf")
            Using image As Aspose.CAD.Image = Aspose.CAD.Image.Load(sourceFilePath)
                Dim cadImage As Aspose.CAD.FileFormats.Cad.CadImage = DirectCast(image, Aspose.CAD.FileFormats.Cad.CadImage)

                Dim layouts As Aspose.CAD.FileFormats.Cad.CadLayoutDictionary = cadImage.Layouts
                For Each layout As Aspose.CAD.FileFormats.Cad.CadObjects.CadLayout In layouts.Values
                    Console.WriteLine("Layout " + layout.LayoutName)
                Next
            End Using
            ' ExEnd:ListLayouts
        End Sub
    End Class
End Namespace

File: Examples/Src/ConvertingCAD/ListLayouts.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Aspose.CAD;
namespace Aspose.CAD.Examples.CSharp.ConvertingCAD
{
    public class ListLayouts
    {
        public static void Run()
        {
            //ExStart:ListLayouts
            // The path to the documents directory.
            string MyDir = RunExamples.GetDataDir_ConvertingCAD();
            string sourceFilePath = MyDir + "conic_pyramid.dxf";
            using (Aspose.CAD.Image image = Aspose.CAD.Image.Load(sourceFilePath))
            {
                Aspose.CAD.FileFormats.Cad.CadImage cadImage = (Aspose.CAD.FileFormats.Cad.CadImage)image;

                var layouts = cadImage.Layouts;
                foreach (Aspose.CAD.FileFormats.Cad.CadObjects.CadLayout layout in layouts.Values)
                {
                    Console.WriteLine("Layout " + layout.LayoutName);
                }
            }
            //ExEnd:ListLayouts
        }
    }
}
